figures/3.figure3/scripts/figure3.r

```diff
@@ -481,13 +483,121 @@ umap_plot_death_type <- (
             brewer.pal(3, "Dark2")[1],
             brewer.pal(3, "Dark2")[3]
         ))
-
+    # move the legend to the inside of the plot
+    + theme(
+        legend.position=c(.80,.85),
+        legend.text = element_text(size = 18)
+    )
 )
 # remove the title
 umap_plot_death_type <- umap_plot_death_type + theme(plot.title = element_blank())
 umap_plot_death_type
 
 
+# get the background data for the umap plot
+df_background_control <- cell_umap
+df_background_control$group <- "Control"
+df_background_apoptosis <- cell_umap
+df_background_apoptosis$group <- "Apoptosis"
+df_background_pyroptosis <- cell_umap
+df_background_pyroptosis$group <- "Pyroptosis"
+
+# combine the dataframes
+df_background <- rbind(df_background_control, df_background_apoptosis, df_background_pyroptosis)
+
+cell_umap$group <- factor(cell_umap$group, levels = c("Control","Apoptosis","Pyroptosis"))
+unique(cell_umap$group)
+
+# set plot size
+width <- 8
+height <- 8
+options(repr.plot.width = width, repr.plot.height = height)
+cell_umap <- cell_umap %>% sample_frac(1)
+umap_plot_death_type_facet <- (
+    cell_umap %>% ggplot( aes(x = umap_1, y = umap_2))
+
+    + geom_point(
+        data = df_background,
+        color = "lightgray",
+        size = 1,
```

Comment:
this probably should be the same size as the other (right now it looks like there is a grey halo effect around each point)